routemaster/services/ingest.rb

```diff
@@ -21,18 +24,36 @@ def initialize(topic:, event:, queue:)
       end
 
       def call
-        data = Services::Codec.new.dump(@event)
-        @topic.subscribers.each do |s|
-          batch = Models::Batch.ingest(data: data, timestamp: @event.timestamp, subscriber: s)
+        trace_with_newrelic('Custom/Services/ingest') do
+          data = Services::Codec.new.dump(@event)
+          @topic.subscribers.each do |s|
+            trace_with_newrelic("Custom/Services/ingest-#{s.name}") do
+              batch = nil
+
+              trace_with_newrelic('Custom/Services/batch-ingest') do
+                batch = Models::Batch.ingest(data: data, timestamp: @event.timestamp, subscriber: s)
+              end
+
+              job = Models::Job.new(name: 'batch', args: batch.uid, run_at: batch.deadline)
+
+              trace_with_newrelic('Custom/Services/queue-push') do
+                @queue.push(job)
+              end
+
+              trace_with_newrelic('Custom/Services/queue-promote') do
+                @queue.promote(job) if batch.full?
```

Comment:
While `batch.full?` does some Redis calls, I think that it's `@queue.promote(job)` that we want to trace. Maybe the `if` can be extracted to wrap the whole `trace_with_newrelic` block? What do you think?

Reply:
I'm more suspicious about the `Batch#Ingest` because that's called for each subscriber on each new event and the lua script of that runs 9-10 Redis commands. 

But I will change it and make if statement guard tracing block.